### PR TITLE
Cleanup TableStats/PlanStats

### PR DESCRIPTION
**Pull request by mfussenegger**
Some follow-ups to the PlanStats introduction

See commits for details

### CODE REVIEW

File: server/src/main/java/io/crate/planner/optimizer/costs/PlanStats.java [hunk 1]
```diff
@@ -106,62 +108,58 @@ public Stats visitGroupReference(GroupReference group, Void context) {
...
+                || rhsStats.numDocs() == -1
+                || rhsStats.sizeInBytes() == -1) {
+                return new Stats(-1, -1, statsByColumn);
             }
-            return new Stats(numberOfRows, sizeInBytes, Maps.concat(lhsStats.statsByColumn(), rhsStats.statsByColumn()));
+            long numRows = join.joinType() == JoinType.CROSS
+                ? lhsStats.numDocs() * rhsStats.numDocs()
+                // We don't have any cardinality estimates, so just take the bigger table
+                : Math.max(lhsStats.numDocs(), rhsStats.numDocs());
+            return new Stats(
```
Comment (matriv) [inline server/src/main/java/io/crate/planner/optimizer/costs/PlanStats.java:140]: not part of this PR, just a question: If we have LEFT/SEMI/ANTI shouldn't be `lhsStats.numDocs()` and if RIGHT: `rhsStats.numbDocs()`?

**Review by matriv — APPROVED** [review]
Thx! Left a couple of questions.

File: server/src/testFixtures/java/io/crate/testing/SQLExecutor.java [hunk 2]
```diff
@@ -913,7 +915,7 @@ public <T> T plan(String statement) {
     }
 
     public void updateTableStats(Map<RelationName, Stats> stats) {
-        planStats.tableStats().updateTableStats(stats);
+        tableStats.updateTableStats(stats);
```
[the remark this answers is not in the document]
Comment (mfussenegger) [reply server/src/testFixtures/java/io/crate/testing/SQLExecutor.java:918]: No the `TableStats` are responsible for handling the TableStats.

The main update happens here: 

https://github.com/crate/crate/blob/543d31e48f98d779ba74c369564c48aedb51b9f9/server/src/main/java/io/crate/statistics/TransportAnalyzeAction.java#L140

File: server/src/main/java/io/crate/planner/optimizer/costs/PlanStats.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/planner/optimizer/costs/PlanStats.java:140]: Not sure, I think we'll overhaul this logic anyway.
